tenets/codelingo/effective-go/avoid-renaming-imports/codelingo.yaml

```diff
@@ -0,0 +1,33 @@
+funcs:
+  - name: isInvalid
+    type: asserter
+    body: |
+      function(name) {
+        return name !== "_"
+      }
+tenets:
+  - name: avoid-renaming-imports
+    flows:
+      codelingo/docs:
+        title: Avoid Renaming Imports
+        body: |
+          Avoid renaming imports except to avoid a name collision; good package names
+          should not require renaming. In the event of collision, prefer to rename the
+          most local or project-specific import.
+      codelingo/review:
+        comment: |
+          Avoid renaming imports except to avoid a name collision; good package names
+          should not require renaming. In the event of collision, prefer to rename the
+          most local or project-specific import.
+    query: |
+      import codelingo/ast/go
+
+      go.gen_decl(depth = any):
+        @review comment
+        go.import_spec:
+          go.ident:
+            name as impName
```

Comment:
Why not just `name != "_"` ?

Reply:
Also, you're not covering any named imports other than "_"? You should be able to check the import strings - if they both end in the same name, then a rename is okay, otherwise it's an error.

You should also be able to work out which is the most project-specific import.

Reply:
Will change that. I think there was a bug with the inequality operator when I wrote this. The query will only pick up imports that are named. Unnamed imports are ignored. The idea is to match any named import except for "_"

Reply:
@waigani is referring to an instance such as 

import (
    "text/template"
    htemplate "html/template" // this is now imported as htemplate
)

here it is needed that one of the template imports are named since they are both importing 'template' . So if more than one import has the same ending i.e 'template' then you can ignore it in the query.

Reply:
Ah yup. I'll include that in the tenet